app/uk/gov/hmrc/vatapi/config/AppContext.scala

```
@@ -19,50 +19,43 @@ package uk.gov.hmrc.vatapi.config
   //NRS Config
-  lazy val nrsServiceUrl: String = baseUrl("non-repudiation")
-  lazy val xApiKey: String = config.getString(s"$env.access-keys.xApiKey").getOrElse(throw new RuntimeException("X-API-Key is not configured"))
-  lazy val nrsMaxTimeoutMillis: Int = config.getInt(s"$env.microservice.services.non-repudiation.maxTimeout").getOrElse(throw new RuntimeException("NRS maximum timeout is not configured"))
+  lazy val nrsServiceUrl: String = config.baseUrl("non-repudiation")
+  lazy val xApiKey: String = config.getString(s"access-keys.xApiKey")
+  lazy val nrsMaxTimeoutMillis: Int = config.getInt(s"microservice.services.non-repudiation.maxTimeout")
 
-  lazy val featureSwitch: Option[Configuration] = config.getConfig(s"$env.feature-switch")
+  lazy val featureSwitch: String = config.getString(s"feature-switch")
 
-  lazy val auditEnabled: Boolean = config.getBoolean(s"auditing.enabled").getOrElse(true)
-  lazy val authEnabled: Boolean = config.getBoolean(s"$env.microservice.services.auth.enabled").getOrElse(true)
-  lazy val vatAuthEnrolments: VATAuthEnrolments = VATAuthEnrolments(config.getString(s"$env.enrolments.key").getOrElse(throw new RuntimeException("enrolments.key is not configured")),
-    config.getString(s"$env.enrolments.identifier").getOrElse(throw new RuntimeException("identifier is not configured")),
-    config.getString(s"$env.enrolments.auth-rule"))
-  lazy val vatHybridFeatureEnabled = config.getBoolean(s"$env.feature-switch.des.hybrid").getOrElse(false)
+  lazy val auditEnabled: Boolean = config.getBoolean(s"auditing.enabled")
+  lazy val authEnabled: Boolean = config.getBoolean(s"microservice.services.auth.enabled")
+  lazy val vatAuthEnrolments: VATAuthEnrolments = VATAuthEnrolments(config.getString(s"enrolments.key"),
+    config.getString(s"enrolments.identifier"),
+    Some(config.getString(s"enrolments.authRule")))
+  lazy val vatHybridFeatureEnabled = config.getBoolean(s"feature-switch.des.hybrid")
```

This throws an exception when key missing.

---

Yes, these parameters should be there in the conf. If we missed that should throw an exception

---

But the behaviour has changed. This is too risky as there are no regression tests for this.